internal/provider/group_resource.go

```diff
@@ -0,0 +1,364 @@
+package provider
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &groupResource{}
+)
+
+// NewGroupResource is a helper function to simplify the provider implementation.
+func NewGroupResource() resource.Resource {
+	return &groupResource{}
+}
+
+type GroupResourceModelInterface interface {
+	ParseHttpResponse(body []byte) error
+	CreateRequestBody() ([]byte, diag.Diagnostics)
+	GetURL() string
+}
+
+// groupResource is the resource implementation.
+type groupResource struct {
```

Comment:
```suggestion
type GroupResource struct {
```
I would make this exportable. I'm not sure there's a good reason not to, and for consistency, the job resource in the other PR is exportable.